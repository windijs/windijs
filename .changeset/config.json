{
  "$schema": "https://unpkg.com/@changesets/config@2.1.1/schema.json",
  "changelog": "@changesets/cli/changelog",
  "commit": false,
  "fixed": [["@windijs/core", "windijs"]],
  "linked": [],
  "access": "public",
  "baseBranch": "main",
  "updateInternalDependencies": "patch",
  "ignore": []
}
