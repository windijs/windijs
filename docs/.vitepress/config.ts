import { defineConfig } from "vitepress";
import { mdRenderFilename } from "./markdown";

export default defineConfig({
  lang: "en-US",
  title: "Windi JS",
  cleanUrls: "without-subfolders",
  description: "The Utility-first CSS-in-JS Framework Powered by ES6 Proxy API and TypeScript.",
  markdown: {
    config(md) {
      md.use(mdRenderFilename);
    },
  },
  head: [
    ['link', { rel: "shortcut icon", type: "image/x-icon", href: "/windijs/favicon.ico" }]
  ],
  base: "/windijs/",
  themeConfig: {
    logo: "/logo.png",
    nav: [
      { text: "Guide", link: "/guide/what-is-windijs" },
      { text: "API", link: "/api" },
      { text: "Repl", link: "/repl" },
      { text: "Examples", link: "https://github.com/windijs/windijs/tree/main/examples" },
    ],
    sidebar: [
      {
        text: "Guide",
        collapsible: false,
        items: [
          {
            text: "What is Windi JS?",
            link: "/guide/what-is-windijs",
          },
          {
            text: "Getting Started",
            link: "/guide/getting-started",
          },
          {
            text: "Configuration",
            link: "/guide/configuration",
          },
          {
            text: "Optimization",
            link: "/guide/optimization",
          },
        ],
      },
      {
        text: "Customization",
        collapsible: false,
        items: [
          {
            text: "Theme",
            link: "/customization/theme",
          },
          {
            text: "Utility",
            link: "/customization/utility",
          },
          {
            text: "Variant",
            link: "/customization/variant",
          },
          {
            text: "Handler",
            link: "/customization/handler",
          },
          {
            text: "Loader",
            link: "/customization/loader",
          },
          {
            text: "Namer",
            link: "/customization/namer",
          },
          {
            text: "Plugins",
            link: "/customization/plugins",
          },
        ],
      },
      {
        text: "Integrations",
        collapsible: false,
        items: [
          {
            text: "API",
            link: "/integrations/api",
          },
          // {
          //   text: "CLI",
          //   link: "/integrations/cli",
          // },
          {
            text: "Vite",
            link: "/integrations/vite",
          },
          // {
          //   text: "Rollup",
          //   link: "/integrations/rollup",
          // },
          // {
          //   text: "Webpack",
          //   link: "/integrations/webpack",
          // },
        ],
      },
      {
        text: "Posts",
        collapsible: false,
        items: [
          {
            text: "How Windi JS Works",
            link: "/posts/how-windijs-works",
          },
          {
            text: "Ways of Using Windi JS",
            link: "/posts/ways-of-using-windijs",
          },
        ],
      },
    ],
    editLink: {
      pattern: "https://github.com/windijs/windijs/edit/main/docs/:path",
      text: "Edit this page on GitHub",
    },
    socialLinks: [
      { icon: "discord", link: "https://discord.gg/nJux6Gb52m" },
      { icon: "github", link: "https://github.com/windijs/windijs" },
    ],
    footer: {
      message: "Released under the MIT License.",
      copyright: "Copyright Â© 2022-present Raven Satir",
    },
  },
});
