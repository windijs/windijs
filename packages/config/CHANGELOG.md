# @windijs/config

## 1.0.6

### Patch Changes

- [`134427d6`](https://github.com/windijs/windijs/commit/134427d666a61704de77d52f0e6e4ffbfe085cd3) Thanks [@voorjaar](https://github.com/voorjaar)! - fix: export package.json

* [`99ecf595`](https://github.com/windijs/windijs/commit/99ecf5956d7fa1124c1561d8f58b7a0a2f1af0f2) Thanks [@voorjaar](https://github.com/voorjaar)! - build: switch to es2018 for better tree shaking

* Updated dependencies [[`134427d6`](https://github.com/windijs/windijs/commit/134427d666a61704de77d52f0e6e4ffbfe085cd3), [`99ecf595`](https://github.com/windijs/windijs/commit/99ecf5956d7fa1124c1561d8f58b7a0a2f1af0f2)]:
  - @windijs/shared@1.0.5

## 1.0.5

### Patch Changes

- [`b38b5607`](https://github.com/windijs/windijs/commit/b38b5607dccfb2d5b22af15f1b62af9ad9caf3ae) Thanks [@voorjaar](https://github.com/voorjaar)! - feat: complete config define

* [`3bfacd6e`](https://github.com/windijs/windijs/commit/3bfacd6e745c01f650f1f5e7b65574585ee8c5bc) Thanks [@voorjaar](https://github.com/voorjaar)! - fix: important config with string

* Updated dependencies [[`d285c569`](https://github.com/windijs/windijs/commit/d285c569ec7d4cdd0921f92c7dfe892e58c9cd75), [`831cae71`](https://github.com/windijs/windijs/commit/831cae71465e13a4d00188cca914710c54cd29cf)]:
  - @windijs/shared@1.0.4

## 1.0.5-beta.3

### Patch Changes

- [`3bfacd6e`](https://github.com/windijs/windijs/commit/3bfacd6e745c01f650f1f5e7b65574585ee8c5bc) Thanks [@voorjaar](https://github.com/voorjaar)! - fix: important config with string

## 1.0.5-beta.2

### Patch Changes

- [`b38b5607`](https://github.com/windijs/windijs/commit/b38b5607dccfb2d5b22af15f1b62af9ad9caf3ae) Thanks [@voorjaar](https://github.com/voorjaar)! - feat: complete config define

## 1.0.5-beta.1

### Patch Changes

- Updated dependencies [[`d285c569`](https://github.com/windijs/windijs/commit/d285c569ec7d4cdd0921f92c7dfe892e58c9cd75)]:
  - @windijs/shared@1.0.4-beta.1

## 1.0.5-beta.0

### Patch Changes

- Updated dependencies [[`831cae71`](https://github.com/windijs/windijs/commit/831cae71465e13a4d00188cca914710c54cd29cf)]:
  - @windijs/shared@1.0.4-beta.0

## 1.0.4

### Patch Changes

- af3c0c04: feat: add defineConfig interface
- Updated dependencies [939f0398]
  - @windijs/shared@1.0.3

## 1.0.3

### Patch Changes

- fix: should publish CHANGELOG.md
- Updated dependencies
  - @windijs/shared@1.0.2

## 1.0.2

### Patch Changes

- Updated dependencies
  - @windijs/shared@1.0.1

## 1.0.1

### Patch Changes

- 98ef7277: fix: package exports
