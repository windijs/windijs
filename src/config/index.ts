export * from "./colors";
export * from "./bg";
export * from "./flex";
export * from "./grid";
export * from "./border";
export * from "./filter";
export * from "./opacity";
export * from "./sizing";
export * from "./spacing";
export * from "./positioning";
export * from "./interactivity";
export * from "./typography";
