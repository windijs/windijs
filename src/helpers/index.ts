export * from "./css";
export * from "./unit";
export * from "./funcs";
export * from "./namer";
export * from "./build";
