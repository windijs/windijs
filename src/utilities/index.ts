export * from "./base";
export * from "./builder";
export * from "./style";
export * from "./api";
export * from "./bg";
export * from "./font";
export * from "./spacing";
