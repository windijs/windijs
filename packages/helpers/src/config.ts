import { Handler, StyleLoader, StyleObject } from "./types";

import { Theme } from "@windijs/config";

export type DarkModeConfig = "class" | "media" | false;

export type UtilitiesConfig = { [key: string]: Handler<unknown> | Array<Handler<unknown>> | StyleObject | UtilitiesConfig };

export type VariantsConfig = { [key: string]: string | string[] };

export interface Config<T extends object = Theme> {
  // presets?: Config[];
  // prefixer?: boolean;
  // attributify?: boolean | {
  //   prefix?: string
  //   separator?: string
  //   disable?: string[]
  // }
  separator?: string;
  important?: boolean | string;
  darkMode?: DarkModeConfig;
  theme?: T;
  // variantOrder?: string[];
  // plugins?: Plugin[];
  // handlers?: Handlers;
  // corePlugins?: (keyof BaseTheme)[] | string[] | { [ T in keyof BaseTheme ] : boolean } | { [ key:string ] : boolean };
  // prefix?: string;
  // exclude?: RegExp[];
  // alias?: { [key:string]: string };
  // shortcuts?: { [key:string]: Shortcut };
  // purge?: unknown;
  utilities?: UtilitiesConfig;
  variants?: VariantsConfig;
  styleLoader?: StyleLoader;
  [key:string]: any;
}

export function defineConfig (config: Config) {
  return config;
}
