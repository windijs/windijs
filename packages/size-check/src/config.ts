import { paddingConfig } from "@windijs/config";

export { paddingConfig };
