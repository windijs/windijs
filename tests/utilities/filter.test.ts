import { blurConfig, brightnessConfig, contrastConfig, dropShadowConfig, grayscaleConfig, hueRotateConfig, invertConfig, opacityConfig, saturateConfig, sepiaConfig } from "config";
import { buildBackdropFilter, buildFilter, configHandler, createUtility, cssHandler } from "utilities";
import { css, deg, filters, percent, px, rem, rgba } from "helpers";

test("Blur", () => {
  const blur = createUtility("blur")
    .use(configHandler(blurConfig, v => css({ "--w-blur": `blur(${v})` })))
    .init(filters.blur);

  expect(blur.css).toMatchSnapshot();
  expect(blur.sm.css).toMatchSnapshot();
  expect(blur.lg.css).toMatchSnapshot();
  expect(blur(rem[0.5])).toEqual("blur(0.5rem)");
});

test("Brightness", () => {
  const brightness = createUtility("brightness")
    .use(configHandler(brightnessConfig, v => css({ "--w-brightness": `brightness(${v})` })))
    .init(filters.brightness);

  expect(brightness[100].css).toMatchSnapshot();
  expect(brightness[75].css).toMatchSnapshot();
  expect(brightness(0.8)).toEqual("brightness(0.8)");
});

test("Contrast", () => {
  const contrast = createUtility("contrast")
    .use(configHandler(contrastConfig, v => css({ "--w-contrast": `contrast(${v})` })))
    .init(filters.contrast);

  expect(contrast[100].css).toMatchSnapshot();
  expect(contrast[75].css).toMatchSnapshot();
  expect(contrast(0.8)).toEqual("contrast(0.8)");
});

test("Drop Shadow", () => {
  const dropShadow = createUtility("dropShadow")
    .use(configHandler(dropShadowConfig, v => css({ "--w-drop-shadow": Array.isArray(v) ? v.map(i => `drop-shadow(${i})`).join(" ") : `drop-shadow(${v})` })))
    .init(filters.dropShadow);

  expect(dropShadow.css).toMatchSnapshot();
  expect(dropShadow.sm.css).toMatchSnapshot();
  expect(dropShadow.xl.css).toMatchSnapshot();
  expect(dropShadow(0, px[4], px[3], rgba(0, 0, 0, 0.07))).toEqual("drop-shadow(0 4px 3px rgba(0, 0, 0, 0.07))");
});

test("Grayscale", () => {
  const grayscale = createUtility("grayscale")
    .use(configHandler(grayscaleConfig, v => css({ "--w-grayscale": `grayscale(${v})` })))
    .init(filters.grayscale);

  expect(grayscale.css).toMatchSnapshot();
  expect(grayscale[10].css).toMatchSnapshot();
  expect(grayscale[75].css).toMatchSnapshot();
  expect(grayscale(0.5)).toEqual("grayscale(0.5)");
});

test("Hue Rotate", () => {
  const hueRotate = createUtility("hueRotate")
    .use(configHandler(hueRotateConfig, v => css({ "--w-hue-rotate": `hue-rotate(${v})` })))
    .init(filters.hueRotate);

  expect(hueRotate[-30].css).toMatchSnapshot();
  expect(hueRotate[60].css).toMatchSnapshot();
  expect(hueRotate(deg[25])).toEqual("hue-rotate(25deg)");
});

test("Invert", () => {
  const invert = createUtility("invert")
    .use(configHandler(invertConfig, v => css({ "--w-invert": `invert(${v})` })))
    .init(filters.invert);

  expect(invert.css).toMatchSnapshot();
  expect(invert[20].css).toMatchSnapshot();
  expect(invert[50].css).toMatchSnapshot();
  expect(invert(0.35)).toEqual("invert(0.35)");
});

test("Saturate", () => {
  const saturate = createUtility("saturate")
    .use(configHandler(saturateConfig, v => css({ "--w-saturate": `saturate(${v})` })))
    .init(filters.saturate);

  expect(saturate.css).toMatchSnapshot();
  expect(saturate[75].css).toMatchSnapshot();
  expect(saturate[50].css).toMatchSnapshot();
  expect(saturate(0.35)).toEqual("saturate(0.35)");
});

test("Sepia", () => {
  const sepia = createUtility("sepia")
    .use(configHandler(sepiaConfig, v => css({ "--w-sepia": `sepia(${v})` })))
    .init(filters.sepia);

  expect(sepia.css).toMatchSnapshot();
  expect(sepia[20].css).toMatchSnapshot();
  expect(sepia[50].css).toMatchSnapshot();
  expect(sepia(0.35)).toEqual("sepia(0.35)");
});

test("Filter", () => {
  const filter = createUtility("filter")
    .use(cssHandler({
      "--w-blur": "var(--w-empty,/*!*/ /*!*/)",
      "--w-brightness": "var(--w-empty,/*!*/ /*!*/)",
      "--w-contrast": "var(--w-empty,/*!*/ /*!*/)",
      "--w-grayscale": "var(--w-empty,/*!*/ /*!*/)",
      "--w-hue-rotate": "var(--w-empty,/*!*/ /*!*/)",
      "--w-invert": "var(--w-empty,/*!*/ /*!*/)",
      "--w-saturate": "var(--w-empty,/*!*/ /*!*/)",
      "--w-sepia": "var(--w-empty,/*!*/ /*!*/)",
      "--w-drop-shadow": "var(--w-empty,/*!*/ /*!*/)",
      "-webkit-filter": "var(--w-blur) var(--w-brightness) var(--w-contrast) var(--w-grayscale) var(--w-hue-rotate) var(--w-invert) var(--w-saturate) var(--w-sepia) var(--w-drop-shadow)",
      filter: "var(--w-blur) var(--w-brightness) var(--w-contrast) var(--w-grayscale) var(--w-hue-rotate) var(--w-invert) var(--w-saturate) var(--w-sepia) var(--w-drop-shadow)",
    }))
    .init(buildFilter);

  const blur = createUtility("blur")
    .use(configHandler(blurConfig, v => css({ "--w-blur": `blur(${v})` })))
    .init(filters.blur);

  const brightness = createUtility("brightness")
    .use(configHandler(brightnessConfig, v => css({ "--w-brightness": `brightness(${v})` })))
    .init(filters.brightness);

  expect(filter(blur(px[3]), brightness(percent[10])).css).toMatchSnapshot();

  expect(filter(blur.lg, brightness[50]).css).toMatchSnapshot();

  expect([filter.css, blur.lg.css, brightness[75].css]).toMatchSnapshot();
});

test("Backdrop Filter", () => {
  const backdrop = createUtility("backdrop")
    .use(cssHandler({
      "--w-backdrop-blur": "var(--w-empty,/*!*/ /*!*/)",
      "--w-backdrop-brightness": "var(--w-empty,/*!*/ /*!*/)",
      "--w-backdrop-contrast": "var(--w-empty,/*!*/ /*!*/)",
      "--w-backdrop-grayscale": "var(--w-empty,/*!*/ /*!*/)",
      "--w-backdrop-hue-rotate": "var(--w-empty,/*!*/ /*!*/)",
      "--w-backdrop-invert": "var(--w-empty,/*!*/ /*!*/)",
      "--w-backdrop-opacity": "var(--tw-empty,/*!*/ /*!*/)",
      "--w-backdrop-saturate": "var(--w-empty,/*!*/ /*!*/)",
      "--w-backdrop-sepia": "var(--w-empty,/*!*/ /*!*/)",
      "-webkit-backdrop-filter": "var(--w-backdrop-blur) var(--w-backdrop-brightness) var(--w-backdrop-contrast) var(--w-backdrop-grayscale) var(--w-backdrop-hue-rotate) var(--w-backdrop-invert) var(--w-backdrop-opacity) var(--w-backdrop-saturate) var(--w-backdrop-sepia)",
      backdropFilter: "var(--w-backdrop-blur) var(--w-backdrop-brightness) var(--w-backdrop-contrast) var(--w-backdrop-grayscale) var(--w-backdrop-hue-rotate) var(--w-backdrop-invert) var(--w-backdrop-opacity) var(--w-backdrop-saturate) var(--w-backdrop-sepia)",
    }))
    .case("blur", configHandler(blurConfig, v => css({ "--w-backdrop-blur": `blur(${v})` })))
    .case("brightness", configHandler(brightnessConfig, v => css({ "--w-backdrop-brightness": `brightness(${v})` })))
    .case("contrast", configHandler(contrastConfig, v => css({ "--w-backdrop-contrast": `contrast(${v})` })))
    .case("grayscale", configHandler(grayscaleConfig, v => css({ "--w-backdrop-grayscale": `grayscale(${v})` })))
    .case("hueRotate", configHandler(hueRotateConfig, v => css({ "--w-backdrop-hue-rotate": `hue-rotate(${v})` })))
    .case("invert", configHandler(invertConfig, v => css({ "--w-backdrop-invert": `invert(${v})` })))
    .case("opacity", configHandler(opacityConfig, v => css({ "--w-backdrop-opacity": `opacity(${v})` })))
    .case("saturate", configHandler(saturateConfig, v => css({ "--w-backdrop-saturate": `saturate(${v})` })))
    .case("sepia", configHandler(sepiaConfig, v => css({ "--w-backdrop-sepia": `sepia(${v})` })))
    .init(buildBackdropFilter);

  const blur = createUtility("blur")
    .use(configHandler(blurConfig, v => css({ "--w-blur": `blur(${v})` })))
    .init(filters.blur);

  const brightness = createUtility("brightness")
    .use(configHandler(brightnessConfig, v => css({ "--w-brightness": `brightness(${v})` })))
    .init(filters.brightness);

  expect(backdrop(blur(px[3]), brightness(percent[10])).css).toMatchSnapshot();

  expect(backdrop(blur.lg, brightness[50]).css).toMatchSnapshot();

  expect([backdrop.css, backdrop.blur.lg.css, backdrop.brightness[75].css]).toMatchSnapshot();
});
