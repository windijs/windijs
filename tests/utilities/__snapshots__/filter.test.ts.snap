// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Backdrop Filter 1`] = `
Object {
  "-webkit-backdrop-filter": "blur(3px) brightness(10%)",
  "backdropFilter": "blur(3px) brightness(10%)",
}
`;

exports[`Backdrop Filter 2`] = `
Object {
  "-webkit-backdrop-filter": "blur(16px) brightness(.5)",
  "backdropFilter": "blur(16px) brightness(.5)",
}
`;

exports[`Backdrop Filter 3`] = `
Array [
  Object {
    "--w-backdrop-blur": "var(--w-empty,/*!*/ /*!*/)",
    "--w-backdrop-brightness": "var(--w-empty,/*!*/ /*!*/)",
    "--w-backdrop-contrast": "var(--w-empty,/*!*/ /*!*/)",
    "--w-backdrop-grayscale": "var(--w-empty,/*!*/ /*!*/)",
    "--w-backdrop-hue-rotate": "var(--w-empty,/*!*/ /*!*/)",
    "--w-backdrop-invert": "var(--w-empty,/*!*/ /*!*/)",
    "--w-backdrop-opacity": "var(--tw-empty,/*!*/ /*!*/)",
    "--w-backdrop-saturate": "var(--w-empty,/*!*/ /*!*/)",
    "--w-backdrop-sepia": "var(--w-empty,/*!*/ /*!*/)",
    "-webkit-backdrop-filter": "var(--w-backdrop-blur) var(--w-backdrop-brightness) var(--w-backdrop-contrast) var(--w-backdrop-grayscale) var(--w-backdrop-hue-rotate) var(--w-backdrop-invert) var(--w-backdrop-opacity) var(--w-backdrop-saturate) var(--w-backdrop-sepia)",
    "backdropFilter": "var(--w-backdrop-blur) var(--w-backdrop-brightness) var(--w-backdrop-contrast) var(--w-backdrop-grayscale) var(--w-backdrop-hue-rotate) var(--w-backdrop-invert) var(--w-backdrop-opacity) var(--w-backdrop-saturate) var(--w-backdrop-sepia)",
  },
  Object {
    "--w-backdrop-blur": "blur(16px)",
  },
  Object {
    "--w-backdrop-brightness": "brightness(.75)",
  },
]
`;

exports[`Blur 1`] = `
Object {
  "--w-blur": "blur(8px)",
}
`;

exports[`Blur 2`] = `
Object {
  "--w-blur": "blur(4px)",
}
`;

exports[`Blur 3`] = `
Object {
  "--w-blur": "blur(16px)",
}
`;

exports[`Brightness 1`] = `
Object {
  "--w-brightness": "brightness(1)",
}
`;

exports[`Brightness 2`] = `
Object {
  "--w-brightness": "brightness(.75)",
}
`;

exports[`Contrast 1`] = `
Object {
  "--w-contrast": "contrast(1)",
}
`;

exports[`Contrast 2`] = `
Object {
  "--w-contrast": "contrast(.75)",
}
`;

exports[`Drop Shadow 1`] = `
Object {
  "--w-drop-shadow": "drop-shadow(0 1px 2px rgba(0, 0, 0, 0.1)) drop-shadow(0 1px 1px rgba(0, 0, 0, 0.06))",
}
`;

exports[`Drop Shadow 2`] = `
Object {
  "--w-drop-shadow": "drop-shadow(0 1px 1px rgba(0, 0, 0, 0.05))",
}
`;

exports[`Drop Shadow 3`] = `
Object {
  "--w-drop-shadow": "drop-shadow(0 20px 13px rgba(0, 0, 0, 0.03)) drop-shadow(0 8px 5px rgba(0, 0, 0, 0.08))",
}
`;

exports[`Filter 1`] = `
Object {
  "-webkit-filter": "blur(3px) brightness(10%)",
  "filter": "blur(3px) brightness(10%)",
}
`;

exports[`Filter 2`] = `
Object {
  "-webkit-filter": "blur(16px) brightness(.5)",
  "filter": "blur(16px) brightness(.5)",
}
`;

exports[`Filter 3`] = `
Array [
  Object {
    "--w-blur": "var(--w-empty,/*!*/ /*!*/)",
    "--w-brightness": "var(--w-empty,/*!*/ /*!*/)",
    "--w-contrast": "var(--w-empty,/*!*/ /*!*/)",
    "--w-drop-shadow": "var(--w-empty,/*!*/ /*!*/)",
    "--w-grayscale": "var(--w-empty,/*!*/ /*!*/)",
    "--w-hue-rotate": "var(--w-empty,/*!*/ /*!*/)",
    "--w-invert": "var(--w-empty,/*!*/ /*!*/)",
    "--w-saturate": "var(--w-empty,/*!*/ /*!*/)",
    "--w-sepia": "var(--w-empty,/*!*/ /*!*/)",
    "-webkit-filter": "var(--w-blur) var(--w-brightness) var(--w-contrast) var(--w-grayscale) var(--w-hue-rotate) var(--w-invert) var(--w-saturate) var(--w-sepia) var(--w-drop-shadow)",
    "filter": "var(--w-blur) var(--w-brightness) var(--w-contrast) var(--w-grayscale) var(--w-hue-rotate) var(--w-invert) var(--w-saturate) var(--w-sepia) var(--w-drop-shadow)",
  },
  Object {
    "--w-blur": "blur(16px)",
  },
  Object {
    "--w-brightness": "brightness(.75)",
  },
]
`;

exports[`Grayscale 1`] = `
Object {
  "--w-grayscale": "grayscale(100%)",
}
`;

exports[`Grayscale 2`] = `
Object {
  "--w-grayscale": "grayscale(0.1)",
}
`;

exports[`Grayscale 3`] = `
Object {
  "--w-grayscale": "grayscale(0.75)",
}
`;

exports[`Hue Rotate 1`] = `
Object {
  "--w-hue-rotate": "hue-rotate(-30deg)",
}
`;

exports[`Hue Rotate 2`] = `
Object {
  "--w-hue-rotate": "hue-rotate(60deg)",
}
`;

exports[`Invert 1`] = `
Object {
  "--w-invert": "invert(100%)",
}
`;

exports[`Invert 2`] = `
Object {
  "--w-invert": "invert(0.2)",
}
`;

exports[`Invert 3`] = `
Object {
  "--w-invert": "invert(0.5)",
}
`;

exports[`Saturate 1`] = `
Object {
  "--w-saturate": "saturate(0)",
}
`;

exports[`Saturate 2`] = `
Object {
  "--w-saturate": "saturate(.75)",
}
`;

exports[`Saturate 3`] = `
Object {
  "--w-saturate": "saturate(.5)",
}
`;

exports[`Sepia 1`] = `
Object {
  "--w-sepia": "sepia(100%)",
}
`;

exports[`Sepia 2`] = `
Object {
  "--w-sepia": "sepia(0.2)",
}
`;

exports[`Sepia 3`] = `
Object {
  "--w-sepia": "sepia(0.5)",
}
`;
